enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -0,0 +1,363 @@
+latest IPs and TTL information.
+
+#### Delete/Update
+
+##### Regular DNS name
+
+* An OpenShift cluster administrator deletes an EgressFirewall resource for a Namespace containing rule(s) for regular DNS
+name(s) OR updates an EgressFirewall resource for a Namespace and deletes rule(s) containing regular DNS name(s).
+* The OVN-K master will delete the ACL rule(s) corresponding to the EgressFirewall rule(s) containing the regular DNS name(s).
+* The OVN-K master will then check if the same regular DNS names are also used in the EgressFirewall rules in other Namespaces. If
+they are not used, then the OVN-K master will delete the corrresponding ``AdressSet`` for each of the DNS names in the EgressFirewall
+rules. The OVN-K master will also delete the corresponding ``DNSName`` CRs, only if the ``.spec.isregular`` field is set to true and
+the ``.spec.iswildcard`` field is set to false. If both the fields are set to true, then the CR will not be deleted and the
+``.spec.isregular`` field will be set to false.
+
+
+##### Wildcard DNS name
+
+* An OpenShift cluster administrator deletes an EgressFirewall resource for a Namespace containing rule(s) for wildcard DNS name(s)
+OR updates an EgressFirewall resource for a Namespace and deletes rule(s) containing wildcard DNS name(s).
+* The OVN-K master will delete the ACL rule(s) corresponding to the EgressFirewall rule(s) containing the wildcard DNS name(s).
+* The OVN-K master will then check if the same wildcard DNS names are also used in the EgressFirewall rules in other Namespaces. If
+they are not used, then the OVN-K master will delete the corrresponding ``AdressSet`` for each of the DNS names in the EgressFirewall
+rules. The OVN-K master will also delete the corresponding ``DNSName`` CRs, only if the ``.spec.isregular`` field is set to false and
+the ``.spec.iswildcard`` field is set to true. If both the fields are set to true, then the CR will not be deleted and the
+``.spec.iswildcard`` field will be set to false. From the ``.status`` field all the other DNS names' details will be removed and
+only the details of the DNS name will be kept which matches with the Name field of the CR.
+
+
+#### Variation [optional]
+
+
+### API Extensions
+
+The validation of ``DNSName`` field in ``EgressFirewallDestination`` will be updated to accept wildcard DNS names as well.
+It will be updated from ``^([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.?$`` which accepts only regular DNS names to
+``^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.?$``.
+
+````go
+// EgressFirewallDestination is the endpoint that traffic is either allowed or denied to
+type EgressFirewallDestination struct {
+	// ..
+
+	// dnsName is the domain name to allow/deny traffic to. If this is set, cidrSelector must be unset.
+	// +kubebuilder:validation:Pattern=^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.?$
+	DNSName string `json:"dnsName,omitempty"`
+}
+````
+
+The following ``DNSName`` CRD will be added to the ``k8s.ovn.org`` api-group.
```

Comment:
While _we_ are using the CoreDNS plugin with ovn-kubernetes, it's not logically bound to ovn-kubernetes. I think this should be something like `dns.openshift.io`